controllers/componentplan_controller.go

```diff
@@ -104,32 +120,39 @@ func (r *ComponentPlanReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 		return ctrl.Result{Requeue: true}, nil
 	}
 
-	// Set the status as Unknown when no status are available, and add a finalizer
+	// Set the status as Unknown when no status are available.
 	if plan.Status.Conditions == nil || len(plan.Status.Conditions) == 0 {
-		if plan.Spec.Approved {
-			_ = r.PatchCondition(ctx, plan, logger, corev1alpha1.ComponentPlanAppreoved(), corev1alpha1.ComponentPlanWaitInstall())
-		} else {
-			_ = r.PatchCondition(ctx, plan, logger, corev1alpha1.ComponentPlanUnAppreoved(), corev1alpha1.ComponentPlanWaitInstall())
-		}
-		return ctrl.Result{Requeue: true}, nil
+		return ctrl.Result{Requeue: true}, r.PatchCondition(ctx, plan, logger, r.initCondition(plan)...)
 	}
 
 	// Add a finalizer. Then, we can define some operations which should
 	// occur before the ComponentPlan to be deleted.
 	// More info: https://kubernetes.io/docs/concepts/overview/working-with-objects/finalizers
 	if !controllerutil.ContainsFinalizer(plan, corev1alpha1.Finalizer) {
-		logger.Info("Adding Finalizer for ComponentPlan")
+		logger.Info("Try to add Finalizer for ComponentPlan")
 		if ok := controllerutil.AddFinalizer(plan, corev1alpha1.Finalizer); !ok {
+			logger.Info("Finalizer for ComponentPlan has already been added")
 			return ctrl.Result{Requeue: true, RequeueAfter: 3 * time.Second}, nil
 		}
 
 		if err = r.Update(ctx, plan); err != nil {
-			logger.Error(err, "Failed to update ComponentPlan to add finalizer")
+			logger.Error(err, "Failed to update ComponentPlan to add finalizer, will try again later")
 			return ctrl.Result{}, err
 		}
+		logger.Info("Adding Finalizer for ComponentPlan done")
 		return ctrl.Result{}, nil
 	}
 
+	// Get RESTClientGetter for helm stuff
+	getter, err := r.buildRESTClientGetter()
+	if err != nil {
+		logger.Error(err, "Failed to build RESTClientGetter")
+		return ctrl.Result{Requeue: true, RequeueAfter: 10 * time.Second}, err
+	}
+
+	if r.updateLatest(ctx, logger, getter, plan) {
```

Comment:
Skp when current componentplan's installedVersion is the latest helm release revision.  Is that the case ?
 Pls add some comments on this func `r.updateLatest` since this will break the reconcile flow